flycheck.el

```diff
@@ -4297,17 +4297,20 @@ POS defaults to `point'."
 
 (defun flycheck-jump-to-error (error)
   "Go to the location of ERROR."
-  (let ((error-copy (copy-flycheck-error error))
-        (buffer (find-file-noselect (flycheck-error-filename error))))
-    (setf (flycheck-error-buffer error-copy) buffer)
-    (flycheck-jump-in-buffer buffer error-copy)
-    ;; When jumping to an error in another file, it may not have
-    ;; this error available for highlighting yet, so we trigger a check
-    ;; if necessary.
-    (with-current-buffer buffer
-      (unless (seq-contains flycheck-current-errors error-copy 'equal)
-        (when flycheck-mode
-          (flycheck-buffer))))))
+  (let* ((error-copy (copy-flycheck-error error))
+         (filename (flycheck-error-filename error))
+         (buffer (or (and filename (find-file-noselect filename))
+                     (flycheck-error-buffer error))))
+    (when (buffer-live-p buffer)
+      (setf (flycheck-error-buffer error-copy) buffer)
+      (flycheck-jump-in-buffer buffer error-copy)
+      ;; When jumping to an error in another file, it may not have
+      ;; this error available for highlighting yet, so we trigger a check
+      ;; if necessary.
+      (with-current-buffer buffer
+        (unless (seq-contains flycheck-current-errors error-copy 'equal)
+          (when flycheck-mode
+            (flycheck-buffer)))))))
```

Comment:
It would be nice to restrict this bit of code to run only if we've changed buffers.  Otherwise we scan the whole error list at every move, in most cases for nothing.